lib/eex/lib/eex/compiler.ex

```diff
@@ -67,6 +67,14 @@ defmodule EEx.Compiler do
          scope,
          state
        ) do
+    if mark != '=' do
+      message =
+        "the contents of this \"do\" expression won't be output without the \"<%=\" modifier; if this was intentional " <>
+          "(if the block only has side-effects), please move its contents inside the expression"
```

Comment:
```suggestion
        "the contents of this expression won't be output unless the EEx block starts with \"<%=\""
```

I have removed the "do" mention because it can happen in other cases. I also don't want to prescribe any solution for now because I don't know actual cases where this is actually intentional, so I would rather say nothing and wait for cases to appear (if ever). :)